lib/vdsm/storage/lvmdevices.py

```diff
@@ -74,3 +78,38 @@ def _configure_devices_file(enable=True):
     with lvmconf.LVMConfig() as config:
         config.setint("devices", "use_devicesfile", enabled)
         config.save()
+
+
+def _create_system_devices(vgs):
+    """
+    Import devices of provided VGs into LVM devices file.
+    """
+    for vg in vgs:
+        _run_vgimportdevices(vg)
+
+
+def _run_vgimportdevices(vg):
+    """
+    Import underlying devices of provided VG into LVM devices file. Import is
+    done using vgimportdevices command. vgimportdevices takes into account
+    existing lvm filter, so if some devices are excluded by the filter, such
+    devices won't be imported. If the filter is wrong, we may miss some
+    devices. To avoid such situation, set the filter to enable all the devices.
+    """
+    cmd = [_LVM_VGIMPORTDEVICES,
+           vg,
+           "--config",
+           "devices { filter = [\"a|.*|\"] }"]
```

Comment:
When you have to embed " in the string, use ' instead of escaping:

    'devices { filter = ["a|.*|"] }' 

But I think in this case we need to remove the filter argument.

The first time we run this, the devices file may be missing, so this
command will work.

But in the second time the devicesfile exists, so lvm will complain
about the filter. We cannot disable the devices file because:

```
# vgimportdevices --config 'devices { use_devicesfile = 0 filter = ["a|.*|"] }' cs
  Devices file not enabled.
```

So we cannot use --config to disable the filter when configuring the devices file.
We need to remove the filter from lvm configuration before we configure the devices file.

Reply:
Nice, this make things simpler.